packages/flutter/test/widgets/sliver_tree_test.dart

```diff
@@ -883,4 +883,82 @@ void main() {
     expect(find.text('dos'), findsNothing);
     expect(find.text('tres'), findsNothing);
   });
+
+  testWidgets('TreeSliverNode should close all children when collapsed when animation is completed', (WidgetTester tester) async {
+    final TreeSliverController controller = TreeSliverController();
+    final List<TreeSliverNode<String>> tree = <TreeSliverNode<String>>[
+      TreeSliverNode<String>(
+        'First',
+        expanded: true,
+        children: <TreeSliverNode<String>>[
+          TreeSliverNode<String>(
+            'alpha',
+            expanded: true,
+            children: <TreeSliverNode<String>>[
+              TreeSliverNode<String>('uno'),
+              TreeSliverNode<String>('dos'),
+              TreeSliverNode<String>('tres'),
+            ],
+          ),
+          TreeSliverNode<String>('beta'),
+          TreeSliverNode<String>('kappa'),
+        ],
+      ),
+    ];
+
+
+    Widget buildTreeSliver(TreeSliverController controller) {
+      return MaterialApp(
+        home: Scaffold(
+          body: CustomScrollView(
+            shrinkWrap: true,
+            slivers: <Widget>[
+              TreeSliver<String>(
+                tree: tree,
+                controller: controller,
+                toggleAnimationStyle: AnimationStyle(
+                  curve: Curves.easeInOut,
+                  duration: const Duration(milliseconds: 200),
+                ),
+                treeNodeBuilder: (
+                  BuildContext context,
+                  TreeSliverNode<Object?> node,
+                  AnimationStyle animationStyle,
+                ) {
+                  final Widget child = GestureDetector(
+                    key: ValueKey<String>(node.content! as String),
+                    behavior: HitTestBehavior.translucent,
+                    onTap: () => controller.toggleNode(node),
+                    child: TreeSliver.defaultTreeNodeBuilder(
+                      context,
+                      node,
+                      animationStyle,
+                    ),
+                  );
+                  return child;
+                },
+              ),
+            ],
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(buildTreeSliver(controller));
+
+    expect(find.text('alpha'), findsOneWidget);
+    expect(find.text('uno'), findsOneWidget);
+    expect(find.text('dos'), findsOneWidget);
+    expect(find.text('tres'), findsOneWidget);
+
+    // Using runAsync to handle collapse and animations properly.
+    await tester.runAsync(() async {
```

Comment:
Thanks for the update here as well, can you share why runAsync is needed?

Reply:
The reason I needed to use `runAsync` is that when using `pump` or `pumpAndSettle`, the code was being executed synchronously. That's why I couldn't simulate the real behavior in the test before.
Using `runAsync`, the execution happens asynchronously, reflecting the behavior of a real application.
I don't know if this can be seen as bad practice but it was one of the ways that worked for me. The same applies for https://github.com/flutter/packages/pull/7474#discussion_r1801852257

Reply:
I was mostly curious, thank you!